crates/nu-protocol/src/hir.rs

```diff
@@ -508,13 +505,68 @@ impl Unit {
             Unit::Petabyte => {
                 bytes(convert_number_to_u64(&size) * 1024 * 1024 * 1024 * 1024 * 1024)
             }
-            Unit::Second => duration(convert_number_to_i64(&size)),
-            Unit::Minute => duration(60 * convert_number_to_i64(&size)),
-            Unit::Hour => duration(60 * 60 * convert_number_to_i64(&size)),
-            Unit::Day => duration(24 * 60 * 60 * convert_number_to_i64(&size)),
-            Unit::Week => duration(7 * 24 * 60 * 60 * convert_number_to_i64(&size)),
-            Unit::Month => duration(30 * 24 * 60 * 60 * convert_number_to_i64(&size)),
-            Unit::Year => duration(365 * 24 * 60 * 60 * convert_number_to_i64(&size)),
+            Unit::Nanosecond => duration(size.to_bigint().expect("Conversion should never fail.")),
+            Unit::Microsecond => {
+                duration(size.to_bigint().expect("Conversion should never fail.") * 1000)
+            }
+            Unit::Millisecond => {
+                duration(size.to_bigint().expect("Conversion should never fail.") * 1000 * 1000)
+            }
+            Unit::Second => duration(
+                size.to_bigint().expect("Conversion should never fail.") * 1000 * 1000 * 1000,
+            ),
+            Unit::Minute => duration(
+                size.to_bigint().expect("Conversion should never fail.") * 60 * 1000 * 1000 * 1000,
+            ),
+            Unit::Hour => duration(
+                size.to_bigint().expect("Conversion should never fail.")
+                    * 60
+                    * 60
+                    * 1000
+                    * 1000
+                    * 1000,
+            ),
+            Unit::Day => duration(
+                size.to_bigint().expect("Conversion should never fail.")
+                    * 24
+                    * 60
+                    * 60
+                    * 1000
+                    * 1000
+                    * 1000,
+            ),
+            Unit::Week => duration(
+                size.to_bigint().expect("Conversion should never fail.")
+                    * 7
+                    * 24
+                    * 60
+                    * 60
+                    * 1000
+                    * 1000
+                    * 1000,
+            ),
+            // FIXME: Number of days per month should not always be 30.
```

Comment:
I wonder if we should just remove month and year. Thoughts?

Reply:
I think they could still be convenient, for example when you want to get files that were modified in the last year. We could just not care about the number of days in months and year.